unicorn: 2.1.1 -> 2.1.3

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/project/unicorn-engine/versions.


meta.description for unicorn is: Lightweight multi-platform CPU emulator library

meta.homepage for unicorn is: https://www.unicorn-engine.org




###### Updates performed
- Version update

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 2.1.2 in filename of file in /nix/store/xfi5q57g98gdsy697a0jxwpy8m1hhl01-unicorn-2.1.2

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
13 total rebuild path(s)

13 package rebuild(s)

First fifty rebuilds by attrpath

adenum
gef
pwntools
python312Packages.angr
python312Packages.angrcli
python312Packages.angrop
python312Packages.pwntools
python312Packages.qiling
python312Packages.unicorn
python313Packages.pwntools
python313Packages.unicorn
unicorn
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/xfi5q57g98gdsy697a0jxwpy8m1hhl01-unicorn-2.1.2 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A unicorn https://github.com/r-ryantm/nixpkgs/archive/81a981f5d97ea7b4d8d2c34fb874abdc3d839065.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/81a981f5d97ea7b4d8d2c34fb874abdc3d839065#unicorn
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/xfi5q57g98gdsy697a0jxwpy8m1hhl01-unicorn-2.1.2
ls -la /nix/store/xfi5q57g98gdsy697a0jxwpy8m1hhl01-unicorn-2.1.2/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.angr</li>
    <li>python312Packages.angr.dist</li>
    <li>python312Packages.angrcli</li>
    <li>python312Packages.angrcli.dist</li>
    <li>python312Packages.angrop</li>
    <li>python312Packages.angrop.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>adenum</li>
    <li>gef</li>
    <li>pwntools (python312Packages.pwntools)</li>
    <li>pwntools.dist (python312Packages.pwntools.dist)</li>
    <li>python312Packages.qiling</li>
    <li>python312Packages.qiling.dist</li>
    <li>python312Packages.unicorn</li>
    <li>python312Packages.unicorn.dist</li>
    <li>python313Packages.pwntools</li>
    <li>python313Packages.pwntools.dist</li>
    <li>python313Packages.unicorn</li>
    <li>python313Packages.unicorn.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>unicorn</li>
  </ul>
</details>

---

###### Maintainer pings

cc @thoughtpolice for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
unicorn: 2.1.1 -> 2.1.2

wtdcode commented:
Why does it fail to build? Any logs I can access? We are releasing 2.1.3 in next few days and there is a time window to fix the breakage here.

NickCao commented:
> Why does it fail to build? Any logs I can access? We are releasing 2.1.3 in next few days and there is a time window to fix the breakage here.
> 
> [unicorn-engine/unicorn#2108](https://github.com/unicorn-engine/unicorn/issues/2108)

It's due to the new dependency on versioningit

fabaff committed:
unicorn: 2.1.2 -> 2.1.3

Diff: https://github.com/unicorn-engine/unicorn/compare/refs/tags/2.1.2...2.1.3

fabaff committed:
python313Packages.unicorn: refactor

fabaff commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 381926`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.angr</li>
    <li>python312Packages.angr.dist</li>
    <li>python312Packages.angrcli</li>
    <li>python312Packages.angrcli.dist</li>
    <li>python312Packages.angrop</li>
    <li>python312Packages.angrop.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>adenum</li>
    <li>gef</li>
    <li>pwntools (python312Packages.pwntools)</li>
    <li>pwntools.dist (python312Packages.pwntools.dist)</li>
    <li>python312Packages.qiling</li>
    <li>python312Packages.qiling.dist</li>
    <li>python312Packages.unicorn</li>
    <li>python312Packages.unicorn.dist</li>
    <li>python313Packages.pwntools</li>
    <li>python313Packages.pwntools.dist</li>
    <li>python313Packages.unicorn</li>
    <li>python313Packages.unicorn.dist</li>
    <li>unicorn</li>
  </ul>
</details>

fabaff commented:
#388155 will provide support for `angr`.

risicle commented on pkgs/development/python-modules/unicorn/default.nix:
I think you meant to do the inverse of this?

And you can probably include 32bit by using `isx86` instead.

risicle replied:
Actually scratch that, I can't find a single platform `test_i386` works on.

risicle reviewed (approved):
`nixpkgs-review` happy, macos 12 x86_64, nixos x86_64, aarch64-linux. Built `pkgsStatic.unicorn` & `pkgsLLVM.unicorn` on nixos x86_64 & aarch64-linux, `pkgsCross.riscv64.unicorn` on aarch64-linux. `pkgsMusl.unicorn` on nixos x86_64
